nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/util/PageProviderHelper.java

```diff
@@ -204,19 +208,34 @@ public static PageProviderDefinition getPageProviderDefinition(String providerNa
     public static DocumentModel getSearchDocumentModel(CoreSession session, String providerName,
             Map<String, String> namedParameters) {
         PageProviderService pageProviderService = Framework.getService(PageProviderService.class);
-        PageProviderDefinition def = pageProviderService.getPageProviderDefinition(providerName);
+        return getSearchDocumentModel(session, pageProviderService, providerName, namedParameters);
+    }
 
+    /**
+     * Returns a {@link DocumentModel searchDocumentModel} if given {@code providerName} is not null and has a valid
+     * {@link PageProviderDefinition definition}, or given {@code namedParameters} is not empty.
+     * <p/>
+     * {@link PageProviderDefinition Definition} is valid either it has a type or it declares where clause.
+     * 
+     * @since 11.1
+     */
+    public static DocumentModel getSearchDocumentModel(CoreSession session, PageProviderService pps,
```

Comment:
if given -> if the given
or given -> or the given 
is valid either it has a type or it declares where clause -> is valid if either it has a type or if it declares where clause.